symbols/arm7.yml

```diff
@@ -1,88 +1,161 @@
 arm7:
   versions:
-    - EU
-    - NA
-    - JP
+    - EU-ENTRY
+    - NA-ENTRY
+    - JP-ENTRY
+    - EU-WRAM
+    - NA-WRAM
+    - JP-WRAM
+    - EU-SLOW
+    - NA-SLOW
+    - JP-SLOW
```

Comment:
Really not a fan of this. We should keep the untagged NA/EU/JP as-is, as those directly correspond to the binary itself, and its load address encoded in the cartridge header.

I think it would be a good idea to follow the same format we do for the ITCM (see `arm9/itcm.yml`), which is to record both the unmapped load address (NA/EU/JP), and the relevant mapped runtime address (-WRAM/-SLOW). Since the regions mapped to WRAM and main memory are each contiguous blocks, it would also be great if we could create subregions for them, like with the ITCM.

Reply:
So for ITCM the NA/EU/JP is unmapped initial load addr and NA-ITCM/EU-ITCM/JP-ITCM is mapped addr? Got it, I'll update it.

Reply:
Actually should I move the two later areas into separate files like `slow.yml` and `wram.yml`? I'm hesitant to do it this way because if we do this the main `arm7.yml` file is gonna be very small, the entry section only contains 3 functions solely responsible for memory mapping.

Reply:
> So for ITCM the NA/EU/JP is unmapped initial load addr and NA-ITCM/EU-ITCM/JP-ITCM is mapped addr? Got it, I'll update it.

Yep, that's right. So I guess in this case there'd be _three_ different variants (e.g., NA, NA-WRAM, NA-SLOW).

(Side note, maybe we should change "-SLOW" to just "-RAM"? I think that better reflects what it actually is. We can add a description on the meanings of each region in the top-level description block)

> 

Actually should I move the two later areas into separate files like slow.yml and wram.yml? I'm hesitant to do it this way because if we do this the main arm7.yml file is gonna be very small, the entry section only contains 3 functions solely responsible for memory mapping. It would make more intuitive sense if the wram part was inside of the main file imo, and I would even say that it'd be nice to have the slow part with the wram part too, just tagged differently, but we could stick to what's done with itcm as well, that can be more straightforward in a way.

Technically we don't have to split anything out into separate files, I just think it's nice organizationally. It also means each region can have its own top-level description block (e.g., a `wram.yml` can have a description specially for ARM7 WRAM).

WRAM is fairly small, right? So is it safe to say that the majority of arm7 will reside in main memory? Maybe we could keep the entry stuff and main memory stuff in `arm7.yml`, and have a dedicated file for WRAM stuff? But if it would make more sense to keep everything in the same file for now, I think that's okay too. We can always change this later.

Reply:
> (Side note, maybe we should change "-SLOW" to just "-RAM"? I think that better reflects what it actually is. We can add a description on the meanings of each region in the top-level description block)

Ah yeah that's a better name for it.

> WRAM is fairly small, right? So is it safe to say that the majority of arm7 will reside in main memory? Maybe we could keep the entry stuff and main memory stuff in arm7.yml, and have a dedicated file for WRAM stuff? But if it would make more sense to keep everything in the same file for now, I think that's okay too. We can always change this later.

It's true that it's only about 1/3 of the binary, but at the same time it contains some of the most important functions, like the main event loop, the interrupt setters and getters, and the ipc initialization and send/receive functions. In fact for the two arm7 patches I've done so far, in both instances I've never had to even touch the 0x27E0000 area, which makes me think it might be where functions that aren't used a ton are placed. I'll keep everything in one file for now.

Reply:
> Yep, that's right. So I guess in this case there'd be three different variants (e.g., NA, NA-WRAM, NA-SLOW).

I think there's benefits to including NA-ENTRY as a separate variant as well since the area after those entry functions seem to be reused by the game for other things once the map is complete.

Reply:
> I think there's benefits to including NA-ENTRY as a separate variant as well since the area after those entry functions seem to be reused by the game for other things once the map is complete.

This discussion about reuse came up back when we first added the ITCM stuff too. I think the conclusion was that we shouldn't worry about that until we discover what that memory is actually being reused for, and causing a conflict. The advantage of using the plain "NA", etc. is that it'll put all the artifacts for loading symbols in the main "NA", etc. bucket, rather than separate artifact files.

Reply:
> This discussion about reuse came up back when we first added the ITCM stuff too. I think the conclusion was that we shouldn't worry about that until we discover what that memory is actually being reused for, and causing a conflict. The advantage of using the plain "NA", etc. is that it'll put all the artifacts for loading symbols in the main "NA", etc. bucket, rather than separate artifact files.

I see, I'll match what you guys did for ITCM for this then.